packages/builders/src/messages/Assertions.ts

```diff
@@ -73,5 +82,47 @@ export const messagePredicate = z
 				(data.sticker_ids !== undefined && data.sticker_ids.length > 0)
 			);
 		},
-		{ message: 'Messages must have content, embeds, a poll, attachments, components, or stickers' },
+		{ message: 'Messages must have content, embeds, a poll, attachments, components OR stickers' },
 	);
+
+const allTopLevelComponentsPredicate = z
+	.union([
+		basicActionRowPredicate,
+		z.object({
+			type: z.union([
+				// Components v1
+				z.literal(ComponentType.Button),
```

Comment:
These can't exist as top level components, only in ActionRows or as accessory of a Section. SelectMenus can only exist in ActionRows.